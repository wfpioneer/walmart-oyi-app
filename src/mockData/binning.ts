export const mockPallets = [{
  id: 123456,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '12343534',
    itemNbr: 351231,
    quantity: 2
  }
}, {
  id: 345345,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '345345',
    itemNbr: 351231,
    quantity: 2
  }
}];
