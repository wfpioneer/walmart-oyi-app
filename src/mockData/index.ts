
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
