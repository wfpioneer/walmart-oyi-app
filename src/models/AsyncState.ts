export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
