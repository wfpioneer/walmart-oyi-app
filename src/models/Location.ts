export default interface Location {
  id: string;
  type: string;
  name: string;
}
