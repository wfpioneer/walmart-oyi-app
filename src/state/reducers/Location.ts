import {
  Actions,
  HIDE_LOCATION_POPUP,
  SELECT_AISLE,
  SELECT_SECTION,
  SELECT_ZONE,
  SHOW_LOCATION_POPUP
} from '../actions/Location';

interface LocationState {
  selectedZone: {
    id: number;
    name: string;
  };
  selectedAisle: {
    id: number;
    name: string;
  };
  selectedSection: {
    id: number;
    name: string;
  };
  locationPopupVisible: boolean
}

const initialState: LocationState = {
  selectedZone: {
    id: 0,
    name: ''
  },
  selectedAisle: {
    id: 0,
    name: ''
  },
  selectedSection: {
    id: 0,
    name: ''
  },
  locationPopupVisible: false
};

export const Location = (
  state = initialState,
  action: Actions
) : LocationState => {
  switch (action.type) {
    case SELECT_ZONE: {
      return {
        ...state,
        selectedZone: {
          id: action.payload.id,
          name: action.payload.name
        }
      };
    }
    case SELECT_AISLE: {
      return {
        ...state,
        selectedAisle: {
          id: action.payload.id,
          name: action.payload.name
        }
      };
    }
    case SELECT_SECTION: {
      return {
        ...state,
        selectedSection: {
          id: action.payload.id,
          name: action.payload.name
        }
      };
    }
    case SHOW_LOCATION_POPUP:
      return {
        ...state,
        locationPopupVisible: true
      };
    case HIDE_LOCATION_POPUP:
      return {
        ...state,
        locationPopupVisible: false
      };
    default:
      return state;
  }
};
