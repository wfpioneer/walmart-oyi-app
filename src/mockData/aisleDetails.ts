import { AisleItem } from '../models/LocationItem';

export const mockAisles: AisleItem[] = [
  {
    aisleId: 1,
    aisleName: 'Aisle 1',
    sectionCount: 30
  },
  {
    aisleId: 2,
    aisleName: 'Aisle 2',
    sectionCount: 0
  },
  {
    aisleId: 3,
    aisleName: 'Aisle 3',
    sectionCount: 2
  },
  {
    aisleId: 4,
    aisleName: 'Aisle 4',
    sectionCount: 2
  },
  {
    aisleId: 5,
    aisleName: 'Aisle 5',
    sectionCount: 30
  },
  {
    aisleId: 6,
    aisleName: 'Aisle 6',
    sectionCount: 0
  }
];
