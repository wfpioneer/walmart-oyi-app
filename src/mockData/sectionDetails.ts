import { SectionItem } from '../models/LocationItems';

export const mockSections: SectionItem[] = [
  {
    sectionId: 1,
    sectionName: 'Section 1-1'
  },
  {
    sectionId: 2,
    sectionName: 'Section 1-2'
  },
  {
    sectionId: 3,
    sectionName: 'Section 1-3'
  },
  {
    sectionId: 4,
    sectionName: 'Section 1-4'
  },
  {
    sectionId: 5,
    sectionName: 'Section 1-5'
  },
  {
    sectionId: 6,
    sectionName: 'Section 1-6'
  }
];
