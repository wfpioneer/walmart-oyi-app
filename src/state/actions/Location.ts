import Location from '../../models/Location';

export const SET_FLOOR_LOCATIONS = 'LOCATION/SET_FLOOR_LOCATIONS';
export const SET_RESERVE_LOCATIONS = 'LOCATION/SET_RESERVE_LOCATIONS';
export const SET_ITEM_LOC_DETAILS = 'LOCATION/SET_ITEM_LOC_DETAILS';
export const ADD_LOCATION_TO_EXISTING = 'LOCATION/ADD_LOCATION_TO_EXISTING';
export const EDIT_EXISTING_LOCATION = 'LOCATION/EDIT_EXISTING_LOCATION';
export const IS_UPDATING = 'LOCATION/IS_UPDATING';
export const DELETE_LOCATION_FROM_EXISTING = 'LOCATION/DELETE_LOCATION_FROM_EXISTING';
export const RESET_LOCATIONS = 'LOCATION/RESET_LOCATIONS';

export const setItemLocDetails = (itemNbr: number, upcNbr: string) => ({
  type: SET_ITEM_LOC_DETAILS,
  payload: {
    itemNbr,
    upcNbr
  }
});

export const setFloorLocations = (floor: [Location]) => ({
  type: SET_FLOOR_LOCATIONS,
  payload: floor
});

export const setReserveLocations = (reserve: [Location]) => ({
  type: SET_RESERVE_LOCATIONS,
  payload: reserve
});

export const addLocationToExisting = (locationName: string, locationTypeNbr: number, locationArea: string) => ({
  type: ADD_LOCATION_TO_EXISTING,
  payload: {
    locationName,
    locationTypeNbr,
    locationArea
  }
});

export const editExistingLocation = (locationName: string, locationTypeNbr: number, locationArea: string, locIndex: number) => ({
  type: EDIT_EXISTING_LOCATION,
  payload: {
    locationName,
    locationTypeNbr,
    locationArea,
    locIndex
  }
});

export const deleteLocationFromExisting = (locationArea: string, locIndex: number) => ({
  type: DELETE_LOCATION_FROM_EXISTING,
  payload: {
    locationArea,
    locIndex
  }
});

export const isUpdating = (updating: boolean) => ({
  type: IS_UPDATING,
  payload: updating
});

export const resetLocations = () => ({
  type: RESET_LOCATIONS
});