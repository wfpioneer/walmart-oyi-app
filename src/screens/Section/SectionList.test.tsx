import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { NavigationProp, Route } from '@react-navigation/native';
import { AsyncState } from '../../models/AsyncState';
import {
  ClearItemsModal,
  SectionScreen,
  clearAisleApiEffect,
  deleteAisleApiEffect,
  getSectionsApiEffect,
  handleClearModalClose,
  handleModalClose
} from './SectionList';
import { mockSections } from '../../mockData/sectionDetails';
import { ClearLocationTarget } from '../../models/Location';

let navigationProp: NavigationProp<any>;
let routeProp: Route<any>;
const AISLE_ID = 1;
const AISLE_NAME = '1';
const ZONE_NAME = 'CARN';

const defaultAsyncState: AsyncState = {
  isWaiting: false,
  value: null,
  error: null,
  result: null
};

describe('Test Section List', () => {
  it('Renders Section Screen with no-section-message when getSections response is 204', () => {
    const renderer = ShallowRenderer.createRenderer();
    const getSectionsResult = {
      status: 204,
      data: ''
    };
    const getSectionEmptyResponse: AsyncState = {
      isWaiting: false,
      value: null,
      error: null,
      result: getSectionsResult
    };
    renderer.render(
      <SectionScreen
        aisleId={AISLE_ID}
        aisleName={AISLE_NAME}
        zoneName={ZONE_NAME}
        dispatch={jest.fn()}
        getAllSections={getSectionEmptyResponse}
        isManualScanEnabled={false}
        apiStart={0}
        setApiStart={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
        trackEventCall={jest.fn()}
        locationPopupVisible={false}
        displayConfirmation={false}
        setDisplayConfirmation={jest.fn()}
        deleteAisleApi={defaultAsyncState}
        deleteAisleApiStart={0}
        setDeleteAisleApiStart={jest.fn()}
        isClearAisle={false}
        clearAisleApi={defaultAsyncState}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        setClearLocationTarget={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders Section Screen with Data', () => {
    const renderer = ShallowRenderer.createRenderer();
    const getSectionResult = {
      data: mockSections,
      status: 200
    };
    const getSectionSuccess: AsyncState = {
      isWaiting: false,
      value: null,
      error: null,
      result: getSectionResult
    };
    renderer.render(
      <SectionScreen
        aisleId={AISLE_ID}
        aisleName={AISLE_NAME}
        zoneName={ZONE_NAME}
        dispatch={jest.fn()}
        getAllSections={getSectionSuccess}
        isManualScanEnabled={false}
        apiStart={0}
        setApiStart={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
        trackEventCall={jest.fn()}
        locationPopupVisible={false}
        displayConfirmation={false}
        setDisplayConfirmation={jest.fn()}
        deleteAisleApi={defaultAsyncState}
        deleteAisleApiStart={0}
        setDeleteAisleApiStart={jest.fn()}
        isClearAisle={false}
        clearAisleApi={defaultAsyncState}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        setClearLocationTarget={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders Manual Scan Component when isManualScanEnabled is set to true', () => {
    const renderer = ShallowRenderer.createRenderer();

    renderer.render(
      <SectionScreen
        aisleId={AISLE_ID}
        aisleName={AISLE_NAME}
        zoneName={ZONE_NAME}
        dispatch={jest.fn()}
        getAllSections={defaultAsyncState}
        isManualScanEnabled={true}
        apiStart={0}
        setApiStart={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
        trackEventCall={jest.fn()}
        locationPopupVisible={false}
        displayConfirmation={false}
        setDisplayConfirmation={jest.fn()}
        deleteAisleApi={defaultAsyncState}
        deleteAisleApiStart={0}
        setDeleteAisleApiStart={jest.fn()}
        isClearAisle={false}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        setClearLocationTarget={jest.fn()}
        clearAisleApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('Test Get Section Api Response', () => {
  const possibleErrorResults = [
    { errorType: 'timeout', message: 'timeout of 10000ms exceeded' },
    { errorType: 'network', message: 'Network Error' },
    { errorType: '400', message: 'Request Failed with status code 400' },
    { errorType: '424', message: 'Request Failed with status code 424' },
    { errorType: '500', message: 'Request Failed with status code 500' }
  ];

  possibleErrorResults.forEach(errorResult => it(`Renders Error Message when result is ${errorResult.errorType} error`,
    () => {
      const renderer = ShallowRenderer.createRenderer();
      const apiErrorResult: AsyncState = {
        value: null,
        isWaiting: false,
        error: errorResult.message,
        result: null
      };
      renderer.render(
        <SectionScreen
          aisleId={AISLE_ID}
          aisleName={AISLE_NAME}
          zoneName={ZONE_NAME}
          dispatch={jest.fn()}
          getAllSections={apiErrorResult}
          isManualScanEnabled={false}
          apiStart={0}
          setApiStart={jest.fn()}
          navigation={navigationProp}
          route={routeProp}
          useEffectHook={jest.fn()}
          trackEventCall={jest.fn()}
          locationPopupVisible={false}
          displayConfirmation={false}
          setDisplayConfirmation={jest.fn()}
          deleteAisleApi={defaultAsyncState}
          deleteAisleApiStart={0}
          setDeleteAisleApiStart={jest.fn()}
          isClearAisle={false}
          clearLocationTarget={ClearLocationTarget.FLOOR}
          setClearLocationTarget={jest.fn()}
          clearAisleApi={defaultAsyncState}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    }));

  it('Renders loading indicator when waiting for Section Api response', () => {
    const renderer = ShallowRenderer.createRenderer();
    const getSectionIsWaiting: AsyncState = {
      isWaiting: true,
      value: null,
      error: null,
      result: null
    };
    renderer.render(
      <SectionScreen
        aisleId={AISLE_ID}
        aisleName={AISLE_NAME}
        zoneName={ZONE_NAME}
        dispatch={jest.fn()}
        getAllSections={getSectionIsWaiting}
        isManualScanEnabled={false}
        apiStart={0}
        setApiStart={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
        trackEventCall={jest.fn()}
        locationPopupVisible={false}
        displayConfirmation={false}
        setDisplayConfirmation={jest.fn()}
        deleteAisleApi={defaultAsyncState}
        deleteAisleApiStart={0}
        setDeleteAisleApiStart={jest.fn()}
        isClearAisle={false}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        setClearLocationTarget={jest.fn()}
        clearAisleApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
describe('Rendering Remove Aisle responses', () => {
  it('Renders the waiting for response from Remove Aisle', () => {
    const renderer = ShallowRenderer.createRenderer();
    const removeAisleIsWaiting: AsyncState = {
      isWaiting: true,
      value: null,
      error: null,
      result: null
    };
    renderer.render(
      <SectionScreen
        aisleId={AISLE_ID}
        aisleName={AISLE_NAME}
        zoneName={ZONE_NAME}
        dispatch={jest.fn()}
        getAllSections={defaultAsyncState}
        isManualScanEnabled={false}
        apiStart={0}
        setApiStart={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
        trackEventCall={jest.fn()}
        locationPopupVisible={false}
        displayConfirmation={false}
        setDisplayConfirmation={jest.fn()}
        deleteAisleApi={removeAisleIsWaiting}
        deleteAisleApiStart={0}
        setDeleteAisleApiStart={jest.fn()}
        isClearAisle={false}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        setClearLocationTarget={jest.fn()}
        clearAisleApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('Renders the success response from Remove Aisle', () => {
    const renderer = ShallowRenderer.createRenderer();
    const removeAisleResult = {
      status: 204,
      data: ''
    };
    const removeAisleSuccess: AsyncState = {
      isWaiting: false,
      value: null,
      error: null,
      result: removeAisleResult
    };
    renderer.render(
      <SectionScreen
        aisleId={AISLE_ID}
        aisleName={AISLE_NAME}
        zoneName={ZONE_NAME}
        dispatch={jest.fn()}
        getAllSections={defaultAsyncState}
        isManualScanEnabled={false}
        apiStart={0}
        setApiStart={jest.fn()}
        navigation={navigationProp}
        route={routeProp}
        useEffectHook={jest.fn()}
        trackEventCall={jest.fn()}
        locationPopupVisible={false}
        displayConfirmation={false}
        setDisplayConfirmation={jest.fn()}
        deleteAisleApi={removeAisleSuccess}
        deleteAisleApiStart={0}
        setDeleteAisleApiStart={jest.fn()}
        isClearAisle={false}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        setClearLocationTarget={jest.fn()}
        clearAisleApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('Rendering clear Aisle responses', () => {
  it('Renders the preflight view of clearAisleModal', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <ClearItemsModal
        clearAisleApi={defaultAsyncState}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        displayConfirmation={true}
        handleClearItems={jest.fn()}
        isClearAisle={true}
        setClearLocationTarget={jest.fn()}
        setDisplayConfirmation={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Ensures the clearAisleModal does not show when not clear aisle', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <ClearItemsModal
        clearAisleApi={defaultAsyncState}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        displayConfirmation={true}
        handleClearItems={jest.fn()}
        isClearAisle={false}
        setClearLocationTarget={jest.fn()}
        setDisplayConfirmation={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the waiting for response from clear Aisle', () => {
    const renderer = ShallowRenderer.createRenderer();
    const clearAisleIsWaiting: AsyncState = {
      isWaiting: true,
      value: null,
      error: null,
      result: null
    };
    renderer.render(
      <ClearItemsModal
        clearAisleApi={clearAisleIsWaiting}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        displayConfirmation={true}
        handleClearItems={jest.fn()}
        isClearAisle={true}
        setClearLocationTarget={jest.fn()}
        setDisplayConfirmation={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the success response from clear Aisle', () => {
    const renderer = ShallowRenderer.createRenderer();
    const clearAisleResult = {
      status: 204,
      data: ''
    };
    const clearAisleSuccess: AsyncState = {
      isWaiting: false,
      value: null,
      error: null,
      result: clearAisleResult
    };
    renderer.render(
      <ClearItemsModal
        clearAisleApi={clearAisleSuccess}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        displayConfirmation={true}
        handleClearItems={jest.fn()}
        isClearAisle={true}
        setClearLocationTarget={jest.fn()}
        setDisplayConfirmation={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the failed response from clear aisle', () => {
    const renderer = ShallowRenderer.createRenderer();
    const clearAisleFail: AsyncState = {
      isWaiting: false,
      value: null,
      error: { status: 418, message: 'I am a tea pot' },
      result: null
    };
    renderer.render(
      <ClearItemsModal
        clearAisleApi={clearAisleFail}
        clearLocationTarget={ClearLocationTarget.FLOOR}
        displayConfirmation={true}
        handleClearItems={jest.fn()}
        isClearAisle={true}
        setClearLocationTarget={jest.fn()}
        setDisplayConfirmation={jest.fn()}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

describe('Section List externalized function tests', () => {
  const mockDispatch = jest.fn();

  it('tests handleModalClose', () => {
    const mockSetDisplayConfirmation = jest.fn();
    const mockSetDeleteZoneApiStart = jest.fn();

    handleModalClose(mockSetDisplayConfirmation, mockSetDeleteZoneApiStart, mockDispatch);
    expect(mockSetDisplayConfirmation).toBeCalledTimes(1);
    expect(mockSetDeleteZoneApiStart).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledTimes(1);

    mockDispatch.mockClear();
  });

  it('tests handleClearModalClose', () => {
    const mockSetDisplayConfirmation = jest.fn();

    handleClearModalClose(mockSetDisplayConfirmation, mockDispatch);
    expect(mockSetDisplayConfirmation).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledTimes(1);

    mockDispatch.mockClear();
  });

  it('tests getSectionsApiEffect, success', () => {
    const mockTrackEventCall = jest.fn();
    const successApi: AsyncState = {
      ...defaultAsyncState,
      result: {}
    };

    getSectionsApiEffect(successApi, 0, mockTrackEventCall);
    expect(mockTrackEventCall).toBeCalledTimes(1);
  });

  it('tests getSectionsApiEffect, fail', () => {
    const mockTrackEventCall = jest.fn();
    const failApi: AsyncState = {
      ...defaultAsyncState,
      error: {}
    };

    getSectionsApiEffect(failApi, 0, mockTrackEventCall);
    expect(mockTrackEventCall).toBeCalledTimes(1);
  });

  it('tests clearAisleApiEffect, success', () => {
    const mockIsFocused = jest.fn(() => true);
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    navigationProp = { isFocused: mockIsFocused };
    const mockSetDisplayConfirmation = jest.fn();
    const successApi: AsyncState = {
      ...defaultAsyncState,
      result: {}
    };

    clearAisleApiEffect(mockDispatch, navigationProp, successApi, mockSetDisplayConfirmation);

    expect(mockDispatch).toBeCalledTimes(2);
    expect(mockSetDisplayConfirmation).toBeCalledTimes(1);
    expect(mockIsFocused).toBeCalledTimes(1);

    mockDispatch.mockClear();
  });

  it('tests clearAisleApiEffect, fail', () => {
    const mockIsFocused = jest.fn(() => true);
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    navigationProp = { isFocused: mockIsFocused };
    const mockSetDisplayConfirmation = jest.fn();
    const successApi: AsyncState = {
      ...defaultAsyncState,
      error: {}
    };

    clearAisleApiEffect(mockDispatch, navigationProp, successApi, mockSetDisplayConfirmation);

    expect(mockDispatch).toBeCalledTimes(0);
    expect(mockSetDisplayConfirmation).toBeCalledTimes(0);
    expect(mockIsFocused).toBeCalledTimes(1);

    mockDispatch.mockClear();
  });

  it('tests deleteAisleApiEffect, success', () => {
    const mockIsFocused = jest.fn(() => true);
    const mockGoBack = jest.fn();
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    navigationProp = { isFocused: mockIsFocused, goBack: mockGoBack };
    const mockSetDisplayConfirmation = jest.fn();
    const mockSetDeleteAisleApiStart = jest.fn();
    const mockTrackEventCall = jest.fn();
    const successApi: AsyncState = {
      ...defaultAsyncState,
      result: {}
    };

    deleteAisleApiEffect(
      navigationProp, successApi, 0,
      mockSetDeleteAisleApiStart,
      mockSetDisplayConfirmation,
      mockDispatch, mockTrackEventCall
    );

    expect(mockIsFocused).toBeCalledTimes(1);
    expect(mockTrackEventCall).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledTimes(2);
    expect(mockSetDisplayConfirmation).toBeCalledTimes(1);
    expect(mockSetDeleteAisleApiStart).toBeCalledTimes(1);
    expect(mockGoBack).toBeCalledTimes(1);

    mockDispatch.mockClear();
  });

  it('tests deleteAisleApiEffect, fail', () => {
    const mockIsFocused = jest.fn(() => true);
    const mockGoBack = jest.fn();
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    navigationProp = { isFocused: mockIsFocused, goBack: mockGoBack };
    const mockSetDisplayConfirmation = jest.fn();
    const mockSetDeleteAisleApiStart = jest.fn();
    const mockTrackEventCall = jest.fn();
    const successApi: AsyncState = {
      ...defaultAsyncState,
      error: {}
    };

    deleteAisleApiEffect(
      navigationProp, successApi, 0,
      mockSetDeleteAisleApiStart,
      mockSetDisplayConfirmation,
      mockDispatch, mockTrackEventCall
    );

    expect(mockIsFocused).toBeCalledTimes(1);
    expect(mockTrackEventCall).toBeCalledTimes(1);
    expect(mockDispatch).toBeCalledTimes(2);
    expect(mockSetDisplayConfirmation).toBeCalledTimes(1);
    expect(mockSetDeleteAisleApiStart).toBeCalledTimes(1);
    expect(mockGoBack).toBeCalledTimes(0);

    mockDispatch.mockClear();
  });
});
