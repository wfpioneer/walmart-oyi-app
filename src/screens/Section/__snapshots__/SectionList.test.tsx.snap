// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Get Section Api Response Renders Error Message when result is 400 error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <mockConstructor
    color="#e57373"
    name="alert"
    size={40}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.LOCATION.LOCATION_API_ERROR" translation]
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e57373",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "95%",
      }
    }
  >
    <Text>
      [missing "en.GENERICS.RETRY" translation]
    </Text>
  </ForwardRef>
</View>
`;

exports[`Test Get Section Api Response Renders Error Message when result is 424 error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <mockConstructor
    color="#e57373"
    name="alert"
    size={40}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.LOCATION.LOCATION_API_ERROR" translation]
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e57373",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "95%",
      }
    }
  >
    <Text>
      [missing "en.GENERICS.RETRY" translation]
    </Text>
  </ForwardRef>
</View>
`;

exports[`Test Get Section Api Response Renders Error Message when result is 500 error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <mockConstructor
    color="#e57373"
    name="alert"
    size={40}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.LOCATION.LOCATION_API_ERROR" translation]
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e57373",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "95%",
      }
    }
  >
    <Text>
      [missing "en.GENERICS.RETRY" translation]
    </Text>
  </ForwardRef>
</View>
`;

exports[`Test Get Section Api Response Renders Error Message when result is network error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <mockConstructor
    color="#e57373"
    name="alert"
    size={40}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.LOCATION.LOCATION_API_ERROR" translation]
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e57373",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "95%",
      }
    }
  >
    <Text>
      [missing "en.GENERICS.RETRY" translation]
    </Text>
  </ForwardRef>
</View>
`;

exports[`Test Get Section Api Response Renders Error Message when result is timeout error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <mockConstructor
    color="#e57373"
    name="alert"
    size={40}
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.LOCATION.LOCATION_API_ERROR" translation]
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e57373",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "95%",
      }
    }
  >
    <Text>
      [missing "en.GENERICS.RETRY" translation]
    </Text>
  </ForwardRef>
</View>
`;

exports[`Test Get Section Api Response Renders loading indicator when waiting for Section Api response 1`] = `
<ActivityIndicator
  animating={true}
  color="#0071e9"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`Test Section List Renders Manual Scan Component when isManualScanEnabled is set to true 1`] = `
<View>
  <LocationManualScan
    keyboardType="default"
  />
  <LocationHeader
    details="0 [missing \\"en.LOCATION.SECTIONS\\" translation]"
    dispatch={[MockFunction]}
    location="[missing \\"en.LOCATION.AISLE\\" translation] CARN1"
  />
  <FlatList
    ListEmptyComponent={<NoSectionMessage />}
    contentContainerStyle={
      Object {
        "paddingBottom": 100,
      }
    }
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`Test Section List Renders Section Screen with Data 1`] = `
<View>
  <LocationHeader
    details="6 [missing \\"en.LOCATION.SECTIONS\\" translation]"
    dispatch={[MockFunction]}
    location="[missing \\"en.LOCATION.AISLE\\" translation] CARN1"
  />
  <FlatList
    ListEmptyComponent={<NoSectionMessage />}
    contentContainerStyle={
      Object {
        "paddingBottom": 100,
      }
    }
    data={
      Array [
        Object {
          "sectionId": 1,
          "sectionName": "Section ABAR1-1",
        },
        Object {
          "sectionId": 2,
          "sectionName": "Section ABAR1-2",
        },
        Object {
          "sectionId": 3,
          "sectionName": "Section ABAR1-3",
        },
        Object {
          "sectionId": 4,
          "sectionName": "Section ABAR1-4",
        },
        Object {
          "sectionId": 5,
          "sectionName": "Section ABAR1-5",
        },
        Object {
          "sectionId": 6,
          "sectionName": "Section ABAR1-6",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`Test Section List Renders Section Screen with no-section-message when getSections response is 204 1`] = `
<View>
  <LocationHeader
    details="0 [missing \\"en.LOCATION.SECTIONS\\" translation]"
    dispatch={[MockFunction]}
    location="[missing \\"en.LOCATION.AISLE\\" translation] CARN1"
  />
  <FlatList
    ListEmptyComponent={<NoSectionMessage />}
    contentContainerStyle={
      Object {
        "paddingBottom": 100,
      }
    }
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
